[dnf5] Fix all license headers

dmach opened the pull request:
I still need to re-test this with a tool (a new one I'm going to write) to make sure the changes are consistent.

Conan-Kudo commented:
I'm not really a fan of replacing the headers with these terse SPDX identifiers. They remove the primary means of conveying the intent and spirit of FOSS in the code itself. Maybe I'm the only one that cares about that, but I think it's a terrible idea to remove that information from the code.

dmach commented:
@Conan-Kudo thanks for your feedback, that was exactly what I was looking for.
I wasn't sure about removing the license headers, but I was also hoping that we'll simplify them and reduce the maintenance cost.

I'll do another iteration with the traditional license headers, but with a different Copyright text.
I want to avoid spamming the headers with too many Copyright notes and changing the dates any time a file changes.
I'll probably follow https://www.linuxfoundation.org/en/blog/copyright-notices-in-open-source-software-projects/ and replace all Copyright notes with `Copyright contributors to the libdnf project`.

Stay tuned...

lukash commented:
FWIW would it convey the spirit well enough if there were just a single line like:

> This file is free software, distributed under the GNU General Public License.

(But getting rid of the years that need to be updated is the major point here, the rest is marginal from practical maintenance standpoint)

Conan-Kudo commented:
Some time ago, I asked @richardfontana if we could drop copyright years for Pagure, and he said it was fine. I expect we can do that here too.

richardfontana commented:
The LF article linked by @dmach is pretty much in line with Red Hat Legal's latest recommendations on copyright notices for Red Hat-maintained projects, and it recommends dispensing with dates in copyright notices.

Commit:
Fix all license headers

dmach commented:
All right, another attempt - instead of SPDX headers I've fixed the current headers.
It's a mix of GPLv2+ and LGPLv2.1+.
All copyrights now refer to `Contributors to the libdnf project.`
I've removed copyright year - it has no value and is difficult to maintain.

I've also changed license of libdnf/utils/bgettext/* from LGPLv3+ to LGPLv2.1+.
I think it's ok that I do that on behalf of jrohel, since he's a Red Hat employee, member of my team and he wrote the code during the working hours...

Conan-Kudo commented:
I'll let the tests run, then merge it anyway, since I don't expect the test results to change from changing the license headers...
